### ISSUE TITLE
ULID can't be serialized with Pydantic's dump model to json

### ISSUE THREAD

**Issue by Avihais12344** [mdomke/python-ulid#20]
Hello,
I see this package has support for [`Pydantic`](https://docs.pydantic.dev/latest/). I want to use the [`model_dump_json`](https://docs.pydantic.dev/latest/concepts/serialization/#modelmodel_dump_json) so I could serialize my model to json in my webserver. But I can't serialize ULID. It seems like an implementation error.

How to reproduce:

- Pip freeze:

  ```python
  annotated-types==0.6.0
  pydantic==2.6.4
  pydantic_core==2.16.3
  python-ulid==2.3.0
  typing_extensions==4.10.0
  ```
- Code:

  ```python
  from pydantic import BaseModel
  from pydantic_core import PydanticSerializationError
  from ulid import ULID
  
  
  class MyModel(BaseModel):
      my_id: ULID
  
  my_test = MyModel(my_id=ULID())
  
  try:
      print(f"{my_test.model_dump_json()=}")
  except PydanticSerializationError as e:
      print(f"{e=}")
  
  print(f"{my_test.model_dump()=}")
  ```
  
  The output would be:
  
  ```python
  e=PydanticSerializationError(Error serializing to JSON: PydanticSerializationError: Unable to serialize unknown type: <class 'ulid.ULID'>)
  my_test.model_dump()={'my_id': ULID(01HT4RBZKNQ9DFXZQ8DW02Z27Y)}
  ```

  We can see ha model dump is supported, but not model dump json.

**Comment by Avihais12344** [comment]
A bypass to this problem is to build your own [`field_serializer`](https://docs.pydantic.dev/latest/api/functional_serializers/#pydantic.functional_serializers.field_serializer):

```python
class MyModel(BaseModel):
    my_id: ULID
    
    @field_serializer("my_id", return_type="str", when_used="unless-none")
    @staticmethod
    def serialize_ulid(ulid: ULID) -> str:
        return str(ulid)
```

Then, the code above would work. But I would want support from this library so I wouldn't need to copy-paste it.

**Comment by Avihais12344** [comment]
Opened a PR to solve this issue: https://github.com/mdomke/python-ulid/pull/21

**Comment by orfisko** [comment]
Sorry but this is a major defect. You will get a ton of extra users by having Pydantic support, so serialization should be supported. I just lost an hour because I suspected I did something wrong in my dependencies. Than finding out that this basic feature is not supported is disappointing.

**Comment by mdomke** [comment]
This should be fixed in version `2.4.0` based on the pull-request from @Avihais12344
